lib/ide/src/view/text_editor.rs

```diff
@@ -0,0 +1,118 @@
+//! This module contains TextEditor, an UiComponent to edit Enso Modules or Text Files.
+
+use crate::prelude::*;
+
+use super::ui_component::UiComponent;
+use super::ui_component::Padding;
+
+use basegl::display::object::DisplayObjectOps;
+use basegl::display::shape::text::glyph::font::FontRegistry;
+use basegl::display::shape::text::text_field::TextField;
+use basegl::display::shape::text::text_field::TextFieldProperties;
+use basegl::display::world::*;
+
+use nalgebra::Vector2;
+use nalgebra::zero;
+
+
+
+// =============
+// === Color ===
+// =============
+
+//TODO[dg]:Move this to a better place.
+mod color {
+    use nalgebra::Vector4;
+
+    pub type Color = Vector4<f32>;
+
+    pub fn black() -> Color { Vector4::new(0.0, 0.0, 0.0, 1.0) }
+}
```

Comment:
Do not merge that. Let's clean it up. We are using other Color implementation already. See `core/data/color.rs`.

Reply:
I removed this. I created a TODO comment in `text_field.rs` pointing out that we should change `Vector4<f32>` to the definitions in `core/data/color`.